src/main/java/org/jabref/logic/formatter/minifier/TruncateFormatter.java

```diff
@@ -0,0 +1,47 @@
+package org.jabref.logic.formatter.minifier;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.cleanup.Formatter;
+
+public class TruncateFormatter extends Formatter {
+    public static final String KEY = "truncate";
```

Comment:
I don't understand if their is an advantage of making the key `public static final` rather than  `private final` as it is used for both `equals` and `hashcode`? Setting the `KEY = "truncate" + truncateIndex` as it is already handled as a separate case in `Formatters.getFormatterForModifier`(highlighted below).

Reply:
It should be private. If it's not used outside the class.
Private final static is a compile constant. The compiler replaces all occurrences of the constant with its value. 



And you already have a getter for access for better 
https://www.geeksforgeeks.org/encapsulation-in-java/